### PR TITLE
debugging categories post and get method

### PR DESCRIPTION
**Pull request by msm-faraj**
this pr aims to debug categories get and post method to make them restful

### CODE REVIEW

File: src/contoroller/category/category.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ class CategoryController {
       },
     });
     if (existingCategory) {
-      return res.send("this category has been defined");
+      return res.statu(409).send("this category has been defined");
```
Comment (pouyapourbaba) [inline src/contoroller/category/category.js:27]: Use the [http-status](https://www.npmjs.com/package/http-status) library for the status codes.

Comment (pouyapourbaba) [reply src/contoroller/category/category.js:27]: Actually, you do not need to find the category first. If you add the indexes to your table you can depend on it not to create duplicate categories.

try
   create category
catch
  if error returned by sequelize is duplicate error, return 409 with a meaningful message

File: src/contoroller/category/category.js [hunk 2]
```diff
@@ -78,13 +65,13 @@ class CategoryController {
...
   //** OK **//
-  async getSome(req, res) {
-    const someCategories = await this.Category.findAll({
-      where: { type: req.params.type, deletedAt: null },
+  async getAll(req, res) {
+    const allCategories = await this.Category.findAll({
+      where: { userId: req.user.id, type: req.query.type, deletedAt: null },
     });
-    if (!someCategories)
+    if (!allCategories)
```
Comment (pouyapourbaba) [inline src/contoroller/category/category.js:72]: In getAll methods you cannot return 404. You will return a list of categories here, it either has some items in it or it is empty.

File: src/contoroller/category/category.js [hunk 3]
```diff
@@ -78,13 +65,13 @@ class CategoryController {
   }
 
   //** OK **//
-  async getSome(req, res) {
-    const someCategories = await this.Category.findAll({
-      where: { type: req.params.type, deletedAt: null },
+  async getAll(req, res) {
+    const allCategories = await this.Category.findAll({
+      where: { userId: req.user.id, type: req.query.type, deletedAt: null },
```
Comment (pouyapourbaba) [inline src/contoroller/category/category.js:70]: What if `req.query.type` is null?

You need to check for the query param first and if it exists add it to your where clause.

````
const whereClause = { userId: req.user.id, deletedAt: null };

if  (req.query.type) {
  whereCluase.type = req.query.type;
}

````

Comment (msm-faraj) [reply src/contoroller/category/category.js:70]: done

File: src/contoroller/category/category.js (hunk 1 above)
Comment (msm-faraj) [reply src/contoroller/category/category.js:27]: done

**Pull request merged**